app/schema/item.py

```diff
@@ -17,3 +15,48 @@ def construct_state(self):
 
     def get_state_class(self):
         pass
+
+    def validate(self, state):
+        if isinstance(state, self.get_state_class()):
+            is_valid = True
+            for child_state in state.children:
+                child_schema = self.questionnaire.get_item_by_id(child_state.id)
+                child_valid = child_schema.validate(child_state)
```

Comment:
I'm not sure I understand, as far as I can work out child_valid will always be true or false, so this if statement seems unnecessary?